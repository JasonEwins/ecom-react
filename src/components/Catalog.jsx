import React, { Component } from 'react'
import '../css/catalog.css';
import Product from './Product';


// const API = 'https://d34eedf5-6610-465f-8dcc-aed19b21dee4.mock.pstmn.io/api/v1/product'

class Catalog extends Component {

/* fetchProducts = async () => {
      try {
        const response = await fetch(API);
        const inventory = await response.json();
        
        this.setState ({
          productData: inventory
        });
      } catch (error) {
          this.setState({ error });
      }  
    } 

  componentDidMount() {
    this.fetchProducts();
  }
*/

  render() {

    const { addLikedItem } = this.props;
  
    return (
      <div className="catalog-container">
        <div className="product-menu">
          <p>{this.props.productCount} items</p>
          <button className="button__hide-sold" onClick={this.props.hideSoldItems}>Hide sold items</button>
        </div>
        <div className="product-container">
          <Product {...this.props} addLikedItem={addLikedItem}/>
        </div>
      </div>
    )
  }
}

export default Catalog;