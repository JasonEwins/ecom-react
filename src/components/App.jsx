import React, { Component } from 'react';
import '../css/app.css';
import NavBar from './NavBar';
import Catalog from './Catalog';
import inventory from '../data/Inventory';


class App extends Component {

  constructor(props) {
    super(props)
      this.state = {
      // productData: [],
      // error: null,
      mockProducts: [...inventory],
      productCount: null,
      soldItems: [],
      likedItems: [],
      showLikedItems: false
      }
  }

  authenticate(){
    return new Promise(resolve => setTimeout(resolve, 2000))
  }

  componentDidMount() {
    this.authenticate().then(() => {
      const ele = document.getElementById('ipl-progress-indicator')
      if(ele){
        ele.classList.add('available')
        setTimeout(() => {
          ele.outerHTML = ''
        }, 2000)
      }
    })
    const totalItems = this.state.mockProducts.length;
    this.setState ({
      productCount: totalItems
    })
  }

  addLikedItem = (id) => {
    const item = this.state.mockProducts
      .filter(product => product.id === id)
      .map(product => product.title);
    
    // check to see if item has already been liked
    if (!this.state.likedItems.includes(item.toString())) {
      this.state.likedItems.push(...item);
    
      this.setState({
        likedItems: this.state.likedItems  
      });
    }
    else {
      // if item has been liked remove from list
      let index = this.state.likedItems.indexOf(item.toString());
      let removeItems = this.state.likedItems.splice(index, 1)

      this.setState(state => ({
        likedItems: this.state.likedItems
      }))
    }
  }

  handleToggleLikes = (likedItems) => {
    if (likedItems.length) {
      this.setState(state => ({
        showLikedItems: !this.state.showLikedItems
      }));
    }
  }

  hideSoldItems = () => {
    const soldItem = this.state.mockProducts
      .filter(product => product.sold)
      .map(product => product.id)

      let soldItemId = parseInt(soldItem);
      
      let availableItems = this.state.mockProducts.filter(items => {
        return items.id !== soldItemId
      });
    
    this.setState({
      mockProducts: availableItems,
      productCount: availableItems.length
      })
  }

  render() {
    return (
      <div>
        <NavBar {...this.state} handleToggleLikes={this.handleToggleLikes}/>
        <Catalog {...this.state} addLikedItem={this.addLikedItem} hideSoldItems={this.hideSoldItems}/>
      </div>
    );
  }
}

export default App;
