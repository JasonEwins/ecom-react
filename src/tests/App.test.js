import React from 'react';
import { Adapter } from '../setupTest';
import { shallow, configure } from 'enzyme';
import renderer from 'react-test-renderer';
import App from '../components/App';
import Catalog from '../components/Catalog';

describe('App component', () => {
  it('cta reads "hide sold items"', () => {
    const wrapper = shallow(<Catalog />)
    const text = wrapper.find('button').text();
    expect(text).toEqual('Hide sold items');
  });
});


it('matches the snapshot', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

