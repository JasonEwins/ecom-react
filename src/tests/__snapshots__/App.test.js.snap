// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div>
  <header>
    <nav
      className="nav"
    >
      <div>
        <button
          className="like-button"
          onClick={[Function]}
        >
          <i
            className="far fa-thumbs-up fa-lg"
          />
          0
        </button>
        <div
          className="like-list"
        >
          <ul />
        </div>
      </div>
    </nav>
  </header>
  <div
    className="catalog-container"
  >
    <div
      className="product-menu"
    >
      <p>
        8
         items
      </p>
      <button
        className="button__hide-sold"
        onClick={[Function]}
      >
        Hide sold items
      </button>
    </div>
    <div
      className="product-container"
    >
      <div
        className="items-container"
      >
        <div
          id={1}
        >
          <img
            alt="product"
            className="product__img"
            src="https://d2h1pu99sxkfvn.cloudfront.net/b0/3496737/474561229_0tM1X9VwSk/P8.jpg"
          />
          <h3
            className="product__title"
          >
            Vintage Yellow Champion jumper
          </h3>
          <div
            className="product__brand"
          >
            @
            28_Vintage
          </div>
          <div
            className="product-details"
          >
            <div
              className="product__size"
            >
              Mens XL
            </div>
            <div
              className="product__price"
            >
              Â£35
            </div>
            <span
              onClick={[Function]}
            >
              <i
                className="far fa-thumbs-up fa-2x product__like"
              />
            </span>
          </div>
        </div>
        <div
          id={2}
        >
          <img
            alt="product"
            className="product__img"
            src="https://d2h1pu99sxkfvn.cloudfront.net/b0/9178477/493292127_7EU6wiOPy3/P8.jpg"
          />
          <h3
            className="product__title"
          >
            Marc Jacobs frill mini dress
          </h3>
          <div
            className="product__brand"
          >
            @
            andreea_saioc
          </div>
          <div
            className="product-details"
          >
            <div
              className="product__size"
            >
              12
            </div>
            <div
              className="product__price"
            >
              Â£42
            </div>
            <span
              onClick={[Function]}
            >
              <i
                className="far fa-thumbs-up fa-2x product__like"
              />
            </span>
          </div>
        </div>
        <div
          id={3}
        >
          <div
            className="product-sold-container"
          >
            <div
              className="product__sold"
            >
              sold - too
              <span
                aria-label="fire emoji"
                role="img"
              >
                ðŸ”¥
              </span>
            </div>
            <img
              alt="product"
              className="product__img"
              src="https://d2h1pu99sxkfvn.cloudfront.net/b0/2715120/495715165_Gl4MFCg3Yf/P8.jpg"
            />
          </div>
          <h3
            className="product__title"
          >
            Oversized Longline Trench Coat
          </h3>
          <div
            className="product__brand"
          >
            @
            Rainbow_Vintage
          </div>
          <div
            className="product-details"
          >
            <div
              className="product__size"
            >
              10 +
            </div>
            <div
              className="product__price"
            >
              Â£25
            </div>
            <span
              onClick={[Function]}
            >
              <i
                className="far fa-thumbs-up fa-2x product__like"
              />
            </span>
          </div>
        </div>
        <div
          id={4}
        >
          <img
            alt="product"
            className="product__img"
            src="https://d2h1pu99sxkfvn.cloudfront.net/b0/3834866/495492838_oAlvbJTI9n/P8.jpg"
          />
          <h3
            className="product__title"
          >
            Y2K Red Fred Perry Shirt
          </h3>
          <div
            className="product__brand"
          >
            @
            Onimos
          </div>
          <div
            className="product-details"
          >
            <div
              className="product__size"
            >
              Medium
            </div>
            <div
              className="product__price"
            >
              Â£30
            </div>
            <span
              onClick={[Function]}
            >
              <i
                className="far fa-thumbs-up fa-2x product__like"
              />
            </span>
          </div>
        </div>
        <div
          id={5}
        >
          <img
            alt="product"
            className="product__img"
            src="https://d2h1pu99sxkfvn.cloudfront.net/b0/3456304/481006048_jg5a8tXoOU/P8.jpg"
          />
          <h3
            className="product__title"
          >
            Oversized gola jumper
          </h3>
          <div
            className="product__brand"
          >
            @
            Minisworld
          </div>
          <div
            className="product-details"
          >
            <div
              className="product__size"
            >
              Boys XL
            </div>
            <div
              className="product__price"
            >
              Â£30
            </div>
            <span
              onClick={[Function]}
            >
              <i
                className="far fa-thumbs-up fa-2x product__like"
              />
            </span>
          </div>
        </div>
        <div
          id={6}
        >
          <img
            alt="product"
            className="product__img"
            src="https://d2h1pu99sxkfvn.cloudfront.net/b0/2331652/493195246_sZG5SUxnjb/P8.jpg"
          />
          <h3
            className="product__title"
          >
            Crazy print pattern shirt dress
          </h3>
          <div
            className="product__brand"
          >
            @
            Milingona
          </div>
          <div
            className="product-details"
          >
            <div
              className="product__size"
            >
              Large
            </div>
            <div
              className="product__price"
            >
              Â£25
            </div>
            <span
              onClick={[Function]}
            >
              <i
                className="far fa-thumbs-up fa-2x product__like"
              />
            </span>
          </div>
        </div>
        <div
          id={7}
        >
          <img
            alt="product"
            className="product__img"
            src="https://d2h1pu99sxkfvn.cloudfront.net/b0/3512396/486827753_IA74PcO3TG/P8.jpg"
          />
          <h3
            className="product__title"
          >
            Nikaiyo compact film camera
          </h3>
          <div
            className="product__brand"
          >
            @
            dthcameras
          </div>
          <div
            className="product-details"
          >
            <div
              className="product__size"
            >
              Film, 35mm
            </div>
            <div
              className="product__price"
            >
              Â£20
            </div>
            <span
              onClick={[Function]}
            >
              <i
                className="far fa-thumbs-up fa-2x product__like"
              />
            </span>
          </div>
        </div>
        <div
          id={8}
        >
          <img
            alt="product"
            className="product__img"
            src="https://d2h1pu99sxkfvn.cloudfront.net/b0/2648321/494628660_5SXjgOvMsb/P8.jpg"
          />
          <h3
            className="product__title"
          >
            Patagonia hat
          </h3>
          <div
            className="product__brand"
          >
            @
            Fruity_Finds
          </div>
          <div
            className="product-details"
          >
            <div
              className="product__size"
            >
              Small
            </div>
            <div
              className="product__price"
            >
              Â£21
            </div>
            <span
              onClick={[Function]}
            >
              <i
                className="far fa-thumbs-up fa-2x product__like"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
